src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -623,6 +619,9 @@ public void execute() throws GitException, InterruptedException {
 
                     FetchCommand fetch = git.fetch();
                     fetch.setTagOpt(tags ? TagOpt.FETCH_TAGS : TagOpt.NO_TAGS);
+                    if (allRefSpecs.isEmpty() && tags) {
```

Comment:
probably it is a separate fix, but ok

Reply:
I'm not sure I understand this comment, @oleg-nenashev .  That change is required because the behavior of JGit changed from JGit 4.5 (the version previously supported by the plugin) to JGit 4.9.  We could include the transition from JGit 4.5 to JGit 4.10 as a separate pull request, but it is included as a separate commit already (two commits, showing the transition from JGit 4.5 to JGit 4.9 and then from JGit 4.9 to JGit 4.10).

Reply:
> That change is required because the behavior of JGit changed from JGit 4.5 \[…\] to JGit 4.9.

Then seems fine to me. As a general tip, I find it worthwhile to look over my own PR when filing it to check for unintended changes, like gratuitous reindentation. At the same time I try to look for changes which are desired here but which would not make immediate sense to reviewers because there is no obvious link to the PR description. For those cases, I just leave comments in
 the PR—a self-review which makes reviews by others more streamlined.

(If the code itself is confusing, then add comments to the code—I am talking about code any version of which would look unremarkable if seen in isolation but the diff of which could be unexpected.)

Reply:
Thanks for the advice.  I missed the reality that most code review will be done by looking at the differences in the code, without looking at the comments on the commit which introduced the code.  That applies for code review before the merge of the pull request, and code review when someone is maintaining the code later.

The effort placing good commit messages would have been better spent and more useful as comments in the code with the information related to the commit message.  The knowledge is currently recorded, but not in the best place for those who will most benefit from that knowledge.

Reply:
done